Updates for Assets for Sprint 05.26 (#514)
osisoft/OCS-Docs: content/api-reference/assets/asset-search-api.md

Also returned is a list of `Results`. Each result contains  
- The `MatchProperties` - a list of matched property objects. Each `MatchProperties` object contains the matched fields and their values.
- `Id` - Id of the matched asset.
- `TypeId` - asset type Id of the asset. This is null if the asset does not reference an asset type.
- `Name` - asset name.
- `Description` - asset description.
- `ETag` - Version tag.

> - `CreatedDate` - Created date of the asset.

Reviewer: ```suggestion
- `CreatedDate` - Asset creation date.
```

Author: Thanks! Looks good.

| Query String                                 | Description                                                  |
| -------------------------------------------- | ------------------------------------------------------------ |

> | filter[location]=Earth                       | Filter to ensure that only assets that contain a metadata name = **location** and value is **Earth** is returned. |

Reviewer: ```suggestion
| filter[location]=Earth                       | Filter that only returns assets that contain the metadata name = **location** and the value of this metadata is **Earth**. |
```

Author: Thanks! Looks good.

> | filter[location]=Earth&filter[device]=tracer | Filter to ensure that only assets that contain a metadata name = **location** and value is **Earth** and also contains a metadata with name = **device** and value is **tracer** is returned. |

Reviewer: ```suggestion
| filter[location]=Earth&filter[device]=tracer | Filter that only returns assets that contain both of the following metadata. The first metadata name = **location** with **Earth** as the metadata value, and the second metadata is name = **device** with **tracer** as the metadata value. |
```

Author: Thanks.

> | filter[AssetTypeName]=HeaterType             | Filter to ensure that only assets with AssetTypeName HeaterType is returned. |

Reviewer: ```suggestion
| filter[AssetTypeName]=HeaterType             | Filter that only returns assets with an AssetTypeName of "HeaterType." |
```

Author: Thanks

> | filter[status]=Bad                           | Filter to ensure that only assets with a bad status. Status filters can contain "Good", "Bad", Warning" and "Unknown". |

Reviewer: ```suggestion
| filter[status]=Bad                           | Filter that returns only assets with a bad status. Status filters can have the values "Good", "Bad", Warning", and "Unknown". |
```
What does the second sentence mean? Are you just telling them what the valid values might be?

Author: Yup, the second sentence is mainly for that. Perhaps we should have another section for "Status Filtering"?

    }
]
```

## `Asset Autocomplete` 

> Asset autocomplete provides a way for you to query assets and retrieve a list of suggested assets based on your search criteria. The following asset properties are capable of autocomplete:

Reviewer: ```suggestion
Asset autocomplete allows you to query assets and retrieve a list of suggested assets based on your search criteria. The following asset properties are capable of autocomplete:
```

Author: Looks good.

-  Name
-  Description
-  AssetTypeName
-  Metadata

### Request 

```text 
GET api/v1-preview/Tenants/{tenantId}/Namespaces/{namespaceId}/search/Assets/autocomplete?term={term}&facetCount={facetCount}&termCount={termCount}
```

### Parameters  

`string tenantId`  
The tenant identifier

`string namespaceId`  
The namespace identifier

`string term`  
The search term that you want to search for.

[optional] `int facetCount`  

> The maximum number of facet autocompleted items to return. By default, this value is 0, this means you are not interested in suggestions for facets.

Reviewer: ```suggestion
The maximum number of facet autocompleted items to return. The default is 0 which means you will not get any suggestions for facets.
```

Author: Thanks.,

[optional] `int termCount`  

> The maximum number of token autocompleted items to return. By default, this value is 0, this means you are not interested in suggestions for tokens.

Reviewer: ```suggestion
The maximum number of token autocompleted items to return. The default is 0 which means you will not get any suggestions for tokens.
```

Author: Thanks
